include/battle.h

```diff
@@ -700,7 +700,7 @@ struct BattleStruct
     bool8 effectsBeforeUsingMoveDone:1; // Mega Evo and Focus Punch/Shell Trap effects.
     u8 targetsDone[MAX_BATTLERS_COUNT]; // Each battler as a bit.
     u16 overwrittenAbilities[MAX_BATTLERS_COUNT];    // abilities overwritten during battle (keep separate from battle history in case of switching)
-    bool8 allowedToChangeFormInWeather[PARTY_SIZE][2]; // For each party member and side, used by Ice Face.
+    bool8 allowedToChangeFormInWeather[PARTY_SIZE][NUM_BATTLE_SIDES]; // For each party member and side, used by Ice Face.
```

Comment:
I recommend for things like this just use a bitfield. That way you cut down 12 bytes of memory to 2 (1 for each side).

Reply:
I let it pass for now since we gotta do a general `BattleStruct` cleanup anyway, a revamp of https://github.com/rh-hideout/pokeemerald-expansion/pull/2704.